packages/core/strapi/lib/core/loaders/components.js

```diff
@@ -25,18 +23,20 @@ module.exports = async (strapi) => {
         );
       }
 
-      const uid = `${category}.${key}`;
-
-      acc[uid] = Object.assign(schema, {
-        __schema__: _.cloneDeep(schema),
-        uid,
-        category,
-        modelType: 'component',
-        modelName: key,
-        globalId: schema.globalId || _.upperFirst(_.camelCase(`component_${uid}`)),
-      });
-    });
-
-    return acc;
-  }, {});
+      const definition = {
+        schema: Object.assign(schema, {
+          __schema__: _.cloneDeep(schema),
```

Comment:
Wanted to move this also inside `createComponent` but it will affect the component schema file when editing the component